### ISSUE TITLE
The Error in TestSoftPowers

### ISSUE THREAD

**Issue by earthminator** [smorabit/scWGCNA#127]
Hello, I am now using tutorial data (Zhou_2020.rds) to test hdWGCNA.

I was following the steps described in "hdWGCNA in single-cell data" to execute the codes, but it failed at the step "TestSoftPowers". The error is described as below:

> seurat_obj <- TestSoftPowers(
+   seurat_obj,
+   networkType = 'signed' # you can also use "unsigned" or "signed hybrid"
+ )
pickSoftThreshold: will use block size 3539.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3539 of 12639
Error in datk[c(startG:endG), ] = foreach(t = actualThreads, .combine = rbind) %dopar%  : 
  number of items to replace is not a multiple of replacement length
In addition: Warning message:
In mclapply(argsList, FUN, mc.preschedule = preschedule, mc.set.seed = set.seed,  :
  scheduled cores 1, 2, 3, 4, 5, 6, 7, 8 did not deliver results, all values of the jobs will be affected

What is going on with this?

**Comment by leonardjn** [comment]
I ran into an issue at this step as well after running through the workflow a few times on my own data, although I never received this exact error. Instead, the function printed the following message and never completed over a course of a few hours, even when tried on additional data objects (normally this step took 1-15 minutes on my datasets).

> pickSoftThreshold: will use block size 5346.
>  pickSoftThreshold: calculating connectivity for given powers...
>    ..working on genes 1 through 5346 of 8368

To fix it temporarily, I disabled WGCNA multithreading using `disableWGCNAThreads()`. I re-enabled multithreading for the `ConstructNetwork()` step (e.g. `enableWGCNAThreads(nThreads = 8)`) and that completed without error.

I should note, when I had previously run `TestSoftPowers()` with all WGCNA genes contained in the `pickSoftThreshold()` block size, the test completed without any issue. For example:

> pickSoftThreshold: will use block size 5999.
>  pickSoftThreshold: calculating connectivity for given powers...
>    ..working on genes 1 through 5999 of 5999
